daemon/ZilliqaDaemon.cpp

```diff
@@ -72,18 +74,33 @@ ZilliqaDaemon::ZilliqaDaemon(int argc, const char* argv[], std::ofstream& log)
   StartNewProcess();
 }
 
-void ZilliqaDaemon::MonitorProcess(const string& name) {
+void ZilliqaDaemon::MonitorProcess(const string& name,
+                                   const bool startNewByDaemon) {
   if (m_pids[name].empty()) {
     ZilliqaDaemon::LOG(m_log, "Looking for new " + name + " process...");
     vector<pid_t> tmp = ZilliqaDaemon::GetProcIdByName(name);
+    if (tmp.empty()) {
+      if (!startNewByDaemon &&
+          (m_nodeType == "dsguard" || m_nodeType == "normal")) {
+        auto it = m_failedMonitorProcessCount.find(name);
+        if (it == m_failedMonitorProcessCount.end()) {
+          m_failedMonitorProcessCount[name] = 1;
+        } else {
+          m_failedMonitorProcessCount[name]++;
+        }
+        if (m_failedMonitorProcessCount[name] >= MONITORING_FAIL_COUNT) {
+          StartNewProcess(true);
+          m_failedMonitorProcessCount[name] = 0;
```

Comment:
Do you also need to populate `tmp` at this point with the new process ID? So that when it goes to line 98 below, it will report the process ID.

Reply:
No need i think. It will monitor the process in next run of `MonitorProcess` i.e. after 5 sec.
Note that with `StartNewProcess`, the actual process may take some time to get spawned.  So reading for pid can fail if checked immediately.